packages/doenetml-worker-rust/lib-doenetml-core/src/core/state/types/math_expr.rs

```diff
@@ -0,0 +1,273 @@
+            self.math_object, term
+        );
+
+        let math_object = match DoenetMLCore::eval_js(&js_source) {
+            Ok(res) => res,
+            Err(_) => BLANK_MATH_OBJECT.to_owned(),
+        };
+
+        MathExpr { math_object }
+    }
+
+    /// Create a new mathematical expression by multiplying the current expression by `term`.
+    pub fn multiply(&self, factor: MathOrPrimitive) -> MathExpr {
+        let factor = serde_json::to_string(&factor).unwrap();
+
+        let js_source = format!(
+            r#"JSON.parse('{}', serializedComponentsReviver).multiply(JSON.parse('{}', serializedComponentsReviver));"#,
+            self.math_object, factor
+        );
+
+        let math_object = match DoenetMLCore::eval_js(&js_source) {
+            Ok(res) => res,
+            Err(_) => BLANK_MATH_OBJECT.to_owned(),
+        };
+
+        MathExpr { math_object }
+    }
+
+    /// Create a new mathematical expression by dividing the current expression by `term`.
+    pub fn divide(&self, factor: MathOrPrimitive) -> MathExpr {
+        let factor = serde_json::to_string(&factor).unwrap();
+
+        let js_source = format!(
+            r#"JSON.parse('{}', serializedComponentsReviver).divide(JSON.parse('{}', serializedComponentsReviver));"#,
+            self.math_object, factor
+        );
+
+        let math_object = match DoenetMLCore::eval_js(&js_source) {
+            Ok(res) => res,
+            Err(_) => BLANK_MATH_OBJECT.to_owned(),
+        };
+
+        MathExpr { math_object }
+    }
+}
+
+/// An enum of different types of values that can be interpreted as mathematical expressions.
+#[derive(Debug, serde::Serialize)]
+#[serde(untagged)]
+pub enum MathOrPrimitive {
+    Math(MathExpr),
+    Number(f64),
+    Integer(i64),
+    String(String),
+}
+
+/// Parameters for creating a latex string from a `Math`:
+#[derive(Debug)]
+pub struct ToLatexParams {
+    /// If present, then pad numbers with zeros so they have at least
+    /// this many decimal places after the decimal point displayed.
+    pub pad_to_decimals: Option<i32>,
+    /// If present, then pad numbers with zeros so they have at least
+    /// this many total digits displayed.
+    pub pad_to_digits: Option<i32>,
+    /// If true, then display any blanks in the mathematical expression
+    /// as a long underscore.
+    pub show_blanks: bool,
+}
+
+impl Default for ToLatexParams {
+    fn default() -> Self {
+        ToLatexParams {
+            pad_to_decimals: None,
+            pad_to_digits: None,
+            show_blanks: true,
+        }
+    }
+}
+/// Levels of simplification of mathematical expressions.
+#[derive(Debug, Display, Default)]
+#[strum(serialize_all = "camelCase")]
+pub enum MathSimplify {
+    /// No simplification is performed.
+    #[default]
+    None,
+    /// Simplify numbers, such as simplify `1+1` to 2,
+    /// except do not change the order between numerical and non-numerical operands.
```

Comment:
What is a non-numerical operand? Example, please.

Reply:
Wasn't sure how to word this. I hope the example helps.